src/edu/buffalo/cse/jive/finiteStateMachine/models/State.java

```diff
@@ -17,10 +17,16 @@
 public class State {
 
 	private Map<String, ValueExpression> vector;
-	private boolean valid;
+	private Status status;
+	
+	public enum Status{
+		MARKED,
+		VALID,
+		INVALID
+	}

```

Comment:
### Previous Implementation:
The boolean instance variable 'valid' can hold two states. The invalid states were marked as red, and valid states had a neutral representation.

### New Implementation
We have used an enum to introduce 3 statuses. VALID, INVALID works as usual. If the value is MARKED (only EExpression class can set States as MARKED), it will color it as green indicating the shortest in the graph.